src/Spice86.Aeon/Emulator/Video/Dac.cs

```diff
@@ -1,139 +1,122 @@
 namespace Spice86.Aeon.Emulator.Video
 {
+    using Spice86.Shared;
+
     /// <summary>
     /// Emulates the VGA DAC which provides access to the palette.
     /// </summary>
-    public sealed class Dac
+    public class Dac
     {
-        private readonly unsafe uint* palette;
-        private readonly UnsafeBuffer<uint> paletteBuffer = new(256);
-        private int readChannel;
-        private int writeChannel;
-        private byte readIndex;
-        private byte writeIndex;
+        private readonly Rgb[] _palette = new Rgb[256];
+        private int _readChannel;
+        private int _writeChannel;
+        private byte _readIndex;
+        private byte _writeIndex;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="Dac"/> class.
         /// </summary>
         public Dac()
         {
-            unsafe
-            {
-                palette = paletteBuffer.ToPointer();
-            }
-
             Reset();
         }
 
         /// <summary>
         /// Gets the full 256-color palette.
         /// </summary>
-        public ReadOnlySpan<uint> Palette
-        {
-            get
-            {
-                unsafe
-                {
-                    return new ReadOnlySpan<uint>(palette, 256);
-                }
-            }
-        }
+        public ReadOnlySpan<Rgb> Palette => _palette.AsSpan();
 
         /// <summary>
         /// Gets or sets the current palette read index.
         /// </summary>
         public byte ReadIndex
         {
-            get => readIndex;
+            get => _readIndex;
             set
             {
-                readIndex = value;
-                readChannel = 0;
+                _readIndex = value;
+                _readChannel = 0;
             }
         }
+        
         /// <summary>
         /// Gets or sets the current palette write index.
         /// </summary>
         public byte WriteIndex
         {
-            get => writeIndex;
+            get => _writeIndex;
             set
             {
-                writeIndex = value;
-                writeChannel = 0;
+                _writeIndex = value;
+                _writeChannel = 0;
             }
         }
 
         /// <summary>
         /// Reads the next channel in the current color.
         /// </summary>
         /// <returns>Red, green, or blue channel value.</returns>
-        public byte Read()
-        {
-            unsafe
+        public byte Read() {
+            uint color = _palette[_readIndex];
+            _readChannel++;
```

Comment:
I feel it would be more intuitive to increase the read channel at the end and to have a switch between (0, 1, 2), like arrays indexes

Reply:
Done.